lib/dialogs/timer_dialog.dart

```diff
@@ -4,15 +4,41 @@ import 'package:audioplayers/audioplayers.dart';
 import 'package:flutter/foundation.dart' show kDebugMode;
 
 class TimerDialog {
-  static void showTimerDialog(BuildContext context, {int seconds = 5, Function? onComplete}) {
+  static void showTimerDialog(BuildContext context,
+      {int seconds = 5, Function? onComplete}) {
     int remainingSeconds = seconds;
     Timer? timer;
     bool isPaused = false;
     final player = AudioPlayer();
 
-    void playSound(String source) async {
+    void playSound(String source, {Function? onComplete}) async {
       try {
         await player.play(AssetSource(source));
+        player.onPlayerComplete.listen((event) {
```

Comment:
The selected line adds an event listener to the `player.onPlayerComplete` stream. When the audio playback completes, this listener checks if the `onComplete` callback is provided. If it is, the callback is invoked.

```dart
player.onPlayerComplete.listen((event) { [<- only line selection]
  if (onComplete != null) {
    onComplete();
  }
});
```